swap/swap.go

```diff
@@ -344,6 +368,42 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	return err
 }
 
+func (s *Swap) handleConfirmChequeMsg(ctx context.Context, p *Peer, msg *ConfirmChequeMsg) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	cheque := msg.Cheque
+
+	if p.getPendingCheque() == nil {
+		p.logger.Warn("ignoring confirm msg, no pending cheque")
+	}
+
+	if !cheque.Equal(p.getPendingCheque()) {
+		p.logger.Warn("ignoring confirm msg, wrong cheque", "got", cheque, "expected", p.getPendingCheque())
+		return
+	}
+
+	if err := p.setLastSentCheque(cheque); err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	if err := p.setPendingCheque(nil); err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	if err := p.updateBalance(int64(cheque.Honey)); err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	// since more swap traffic might have occurred since this cheque was already sent, we redo the payment threshold check
+	if err := s.checkPaymentThreshold(p); err != nil {
```

Comment:
is it possible that more traffic has occurred when this function is under lock?

Reply:
Not during `handleConfirmChequeMsg`. But traffic might have occurred since the call to `checkPaymentThreshold` in `Add` where this cheque was initially created.